src/libs/NetworkConnection.ts

```diff
@@ -71,53 +83,81 @@ Onyx.connect({
 
 /**
- * Set up the event listener for NetInfo to tell whether the user has
- * internet connectivity or not. This is more reliable than the Pusher
- * `disconnected` event which takes about 10-15 seconds to emit.
+ * Set interval to periodically (re)check backend status.
+ * Because backend unreachability might imply lost internet connection, we need to check internet reachability.
+ * @returns clearInterval cleanup
  */
-function subscribeToNetInfo(): void {
-    // Note: We are disabling the configuration for NetInfo when using the local web API since requests can get stuck in a 'Pending' state and are not reliable indicators for "offline".
-    // If you need to test the "recheck" feature then switch to the production API proxy server.
-    if (!CONFIG.IS_USING_LOCAL_WEB) {
-        // Calling NetInfo.configure (re)checks current state. We use it to force a recheck whenever we (re)subscribe
-        NetInfo.configure({
-            // By default, NetInfo uses `/` for `reachabilityUrl`
-            // When App is served locally (or from Electron) this address is always reachable - even offline
-            // Using the API url ensures reachability is tested over internet
-            reachabilityUrl: `${CONFIG.EXPENSIFY.DEFAULT_API_ROOT}api/Ping`,
-            reachabilityMethod: 'GET',
-            reachabilityTest: (response) => {
+function subscribeToBackendAndInternetReachability(): () => void {
+    const intervalID = setInterval(() => {
+        // Offline status also implies backend unreachability
+        if (isOffline) {
+            return;
+        }
+        // Using the API url ensures reachability is tested over internet
+        fetch(`${CONFIG.EXPENSIFY.DEFAULT_API_ROOT}api/Ping`, {
+            method: 'GET',
+            cache: 'no-cache',
+        })
+            .then((response) => {
                 if (!response.ok) {
                     return Promise.resolve(false);
                 }
                 return response
                     .json()
                     .then((json) => Promise.resolve(json.jsonCode === 200))
                     .catch(() => Promise.resolve(false));
-            },
+            })
+            .then((isBackendReachable: boolean) => {
+                if (isBackendReachable) {
+                    NetworkActions.setIsBackendReachable(true);
+                    return;
+                }
+                checkInternetReachability().then((isInternetReachable: boolean) => {
+                    setOfflineStatus(!isInternetReachable);
+                    setNetWorkStatus(isInternetReachable);
+                    NetworkActions.setIsBackendReachable(false);
+                });
+            })
+            .catch(() => {
+                checkInternetReachability().then((isInternetReachable: boolean) => {
+                    setOfflineStatus(!isInternetReachable);
+                    setNetWorkStatus(isInternetReachable);
+                    NetworkActions.setIsBackendReachable(false);
```

Comment:
Why is this inside the promise for `checkInternetReachability()`? It doesn't do anything with the `isInternetReachable` value.